# Problem Statement 1:
    
Is gender independent of education level? A random sample of 395 people were
surveyed and each person was asked to report the highest education level they
obtained. The data that resulted from the survey is summarized in the following table:
    
Id   | High School |Bachelors |Masters |Ph.d. |Total
----------|----|------|------|-------|-------|
Female|  60  |   54      |46     |  41 |    201
Male   | 40       |    44  |    53   |    57    | 194
Total  | 100    |      98 |     99  |     98  |   395

Question: Are gender and education level dependent at 5% level of significance? In
other words, given the data collected above, is there a relationship between the
gender of an individual and the level of education that they have obtained?


In [1]:
#Null Hypothesis:           The two categorical variables are independent
#Alternative Hypothesis:    The two categorical variables are dependent 

In [2]:
# Lets find expected frequency
import pandas as pd

In [3]:
Male = [40 ,44 ,53 ,57]
Female =[60 ,54 ,46 ,41]
total_count_row = [201,194]
total_count_column = [100,98,99,98]
total_sampel = 395

In [4]:
female = []
male = []
def expected_frequency(total_count_row, total_count_column, total_sampel):
    for i in range(len(total_count_row)):
        for j in range(len(total_count_column)):
            E = (total_count_row[i]*total_count_column[j])/total_sampel
            if i == 0:
                female.append(E)
            else:
                male.append(E)

In [5]:
expected_frequency(total_count_row,total_count_column,total_sampel)
print(female)
print(male)

[50.88607594936709, 49.868354430379746, 50.37721518987342, 49.868354430379746]
[49.11392405063291, 48.131645569620254, 48.62278481012658, 48.131645569620254]


In [6]:
Dataframe = pd.DataFrame(data =[female,male], index = ["female","male"], columns =  ["High School" ,"Bachelors" ,"Masters" ,"Ph.d"])
Dataframe

High School  Bachelors    Masters       Ph.d
female    50.886076  49.868354  50.377215  49.868354
male      49.113924  48.131646  48.622785  48.131646

In [7]:
def sum_of_chi_square():
    i = 0
    chi_square_new = 0
    for j in range(len(Female)):
        chi_square = (Female[j] - female[j])**2 / female[j]
        chi_square_new = chi_square_new + chi_square
    for j in range(len(male)):
        chi_square = (Male[j] - male[j])**2 / male[j]
        chi_square_new = chi_square_new + chi_square
    return "chi square test: ", chi_square_new

In [8]:
 sum_of_chi_square()

('chi square test: ', 8.006066246262538)

The critical value chi_squre of with 3 degree of freedom is 7.815. Since 8.006 > 7.815, we reject the null hypothesis and conclude that the education level depends on gender at a 5% level of significance.

# Problem Statement 2:
    
Using the following data, perform a oneway analysis of variance using α=.05. Write
up the results in APA format.

[Group1: 51, 45, 33, 45, 67]

[Group2: 23, 43, 23, 43, 45]

[Group3: 56, 76, 74, 87, 56]

In [9]:
import numpy as np
Group1 = [51, 45, 33, 45, 67]
Group2 = [23, 43, 23, 43, 45]
Group3 = [56, 76, 74, 87, 56]
Mean1 = np.mean(Group1)
Mean2 = np.mean(Group2)
Mean3 = np.mean(Group3)
print("Mean of Group1: ", Mean1)
print("Mean of Group2: ", Mean2)
print("Mean of Group3: ", Mean3)

Mean of Group1:  48.2
Mean of Group2:  35.4
Mean of Group3:  69.8


In [10]:
def sum_squared_devation(group, mean):           # function to find Sum of squared deviations from the mean (SS) for the groups:
    sum_sqr_devation = 0
    for i in range(len(group)):
        deveation = group[i]- mean
        devation_sqr = deveation**2
        sum_sqr_devation = sum_sqr_devation + devation_sqr
    return sum_sqr_devation
Group1_sum_sqr_deveation = sum_squared_devation(Group1, Mean1)   #Sum of squared deviations from the mean (SS) for the groups:
Group2_sum_sqr_deveation = sum_squared_devation(Group2, Mean2)   #Sum of squared deviations from the mean (SS) for the groups:  
Group3_sum_sqr_deveation = sum_squared_devation(Group3, Mean3)   #Sum of squared deviations from the mean (SS) for the groups:
print("Group1 sum sqr deveation: ", Group1_sum_sqr_deveation)
print("Group2 sum sqr deveation: ", Group2_sum_sqr_deveation)
print("Group3 sum sqr deveation: ", Group3_sum_sqr_deveation)

Group1 sum sqr deveation:  612.8
Group2 sum sqr deveation:  515.2
Group3 sum sqr deveation:  732.8


In [11]:
Var1 = Group1_sum_sqr_deveation/(len(Group1)-1)
Var2 = Group2_sum_sqr_deveation/(len(Group2)-1)
Var3 = Group3_sum_sqr_deveation/(len(Group3)-1)
print("Variance 1: ",Var1)
print("Variance 2: ",Var2)
print("Variance 3: ",Var3)
MS_error= Var1 + Var2 + Var3
print("Mean squared error: ", MS_error)
#Note: this is just the average within-group variance; it is not sensitive to group mean differences!

#Calculating the remaining error (or within) terms for the ANOVA table:
dferror= (len(Group1)+ len(Group2)+ len(Group3))- 3    # 3 = no_of_group 
SSerror=MS_error * dferror
print("Df error: ", dferror)
print("SS error: ", SSerror)

#Intermediate steps in calculating the variance of the sample means:

Grand_mean = (Mean1+ Mean2+Mean3)/3
print("Grand mean: ", Grand_mean)

Variance 1:  153.2
Variance 2:  128.8
Variance 3:  183.2
Mean squared error:  465.2
Df error:  12
SS error:  5582.4
Grand mean:  51.133333333333326


In [12]:
Mean_Group = [Mean1,Mean2,Mean3]
Grand_mean_sum_sqr_deveation = sum_squared_devation(Mean_Group, Grand_mean)

Var_means = Grand_mean_sum_sqr_deveation/(len(Mean_Group)-1)
MSbetween= Grand_mean_sum_sqr_deveation*(5)

#Note: This method of estimating the variance IS sensitive to group mean differences!

#Calculating the remaining between (or group) terms of the ANOVA table:

dfgroups = len(Mean_Group)-1

SSgroup= MSbetween* dfgroups

In [13]:
#Test statistic and critical value

F =MSbetween/MS_error
F

6.498137002006305

In [14]:
#Fcritical(2,12)=3.89

#So this means we reject null hypothesis

In [15]:
#Effect size
η2 = SSgroup/(SSerror+SSgroup)
η2
#APA writeup

#F(2, 12)=9.75, p <0.05, η2=0.52.

0.5199284502132734

# Problem Statement 3:

Calculate F Test for given 10, 20, 30, 40, 50 and 5,10,15, 20, 25.
For 10, 20, 30, 40, 50:

In [16]:
import numpy as np
Inputs1 =[10,20,30,40,50]
Total_no_sampel = 5
mean1 = np.mean(Inputs1)                                 #calculating mean for frist Data set
Standard_Deviation1 = np.std(Inputs1)                    #calculating standard deviation for frist Data set
Variance1 = np.var(Inputs1)                              #calculating variance for first data set
print("Mean: ", mean1)
print("Standard Deviation: ", Standard_Deviation1)
print("Variance: ", Variance1)

Mean:  30.0
Standard Deviation:  14.142135623730951
Variance:  200.0


In [17]:
Inputs2 =[5,10,15,20,25]
Total_no_sampe2 = 5
mean2 = np.mean(Inputs2)                                 #calculating mean for Second Data set
Standard_Deviation2 = np.std(Inputs2)                    #calculating standard deviation for second Data set
Variance2 = np.var(Inputs2)                              #calculating variance for second data set
print("Mean: ", mean2)
print("Standard Deviation: ", Standard_Deviation2)
print("Variance: ", Variance2)

Mean:  15.0
Standard Deviation:  7.0710678118654755
Variance:  50.0


In [18]:
#  To calculate F Test

f_test = Variance1/Variance2
print("F Test: ",f_test)

F Test:  4.0
